File: src/main/java/me/dreamhopping/pml/api/events/impl/client/net/ClientDisconnectedEvent.java
package me.dreamhopping.pml.api.events.impl.client.net;

import me.dreamhopping.pml.events.Event;

/**
 * Fired when the client is disconnected from the server
 * This is client sided only, for server:
 *
 * @see me.dreamhopping.pml.api.events.impl.server.net.ServerDisconnectEvent
 */
public class ClientDisconnectedEvent extends Event {
    public final String reason;

    /**
     * @param reason: The reason the client was disconnected from the server
     */
    public ClientDisconnectedEvent(String reason) {
        this.reason = reason;
    }
}

File: src/main/java/me/dreamhopping/pml/api/events/impl/client/net/ClientDisconnectedEvent.kt
package me.dreamhopping.pml.api.events.impl.client.net

import me.dreamhopping.pml.api.events.Event

/**
 * Fired when the client is disconnected from the server
 *
 * This is client side only, for server, see [me.dreamhopping.pml.api.events.impl.server.net.ServerDisconnectEvent]
 * @param reason the reason the client was disconnected from the server
 */
class ClientDisconnectedEvent(val reason: String) : Event()